### PR TITLE
Config SSL connection by certificate file path

### PR DESCRIPTION
**Pull request by hchu0701**
To enhance SSL configuration

### CODE REVIEW

**Comment by wzrdtales** [comment]
@hchu0701 Oh ok ic, you reopened that one.

File: index.js [hunk 1]
```diff
@@ -418,6 +419,15 @@ exports.connect = function(config, intern, callback) {
   type = internals.mod.type;
 
   internals.interfaces.SeederInterface._makeParamArgs = dummy;
+  if( config.hasOwnProperty("ssl") && config.ssl.hasOwnProperty("path") && config.ssl.path ){
```
Comment (wzrdtales) [inline index.js]: rather use and check directly against `config.ssl` and `config.ssl.path` for existence than using hasOwnProperty.

File: index.js [hunk 2]
```diff
@@ -418,6 +419,15 @@ exports.connect = function(config, intern, callback) {
...
+    var keys = Object.keys(config.ssl);
+    for(var i=0 ; i < keys.length ; i++ ){
+      var key = keys[i];
+      if( key !== "path" ){
```
Comment (wzrdtales) [inline index.js]: Why do you loop over the whole object if you're just accessing a single object?

**Comment by hchu0701** [comment]
Sorry for the too little description. The following is my request.

If i want a SSL connection, i have to copy all the certificate file contents into database.json or environment variable. So i try to add a parameter "path" meaning all the config.ssl value is a file path. 

I commit a modified practice according to your suggestion. The conversion will happen only when the data type of config.ssl value is a object { "path" : "Your certificate File path" }.

If there is a better way. please let me know.
Thank you.

**Comment by wzrdtales** [comment]
I still don't get why you loop for a simple set operation. What is the reason for this?

File: index.js [hunk 3]
```diff
@@ -418,6 +419,16 @@ exports.connect = function(config, intern, callback) {
   type = internals.mod.type;
 
   internals.interfaces.SeederInterface._makeParamArgs = dummy;
+  if( config.ssl ){
+    var keys = Object.keys(config.ssl);
+    for(var i=0 ; i < keys.length ; i++ ){
+      var key = keys[i];
+      var value = config.ssl[key];
+      if( value instanceof Object && value.path){
+          config.ssl[key] = fs.readFileSync(value.path);
```
Comment (wzrdtales) [inline index.js]: Why would you want to set i.e. `config.ssl.imarandomkeyandiamrandom.path`? And don't just set `config.ssl.path`?

**Comment by hchu0701** [comment]
There might more than one key at `config.ssl`

https://github.com/mysqljs/mysql/blob/eeca1326c5f4d67725a486ddf1ef12c3f12d2187/lib/Connection.js#L288

For example:
```
{
   "host" : "localhost",
   "user" : "USER",
   "password" : "PASSWORD",
   ... 
   "ssl" : {
         "ca" : { "path" : "Your ca file path" },
         "cert" : { "path": "Your cert file path" },
         "key" : { "path": "Your key file path" }
         "ciphers" : "AES128-SHA",
         "passphrase" : "Your Passphrase",
         "rejectUnauthorized" : true
   }
}
```
So i run a loop for converting a object to string from file.
Hope I have understood your question correctly.

**Comment by wzrdtales** [comment]
closing due to old age. can't be merged since it would be a security issue. Only specific keys that exist should be accepted.

**Pull request closed without merging**